build/CMakeLists.txt

```diff
@@ -201,6 +202,49 @@ ExternalProject_Add(prometheus-cpp
     -DCMAKE_INSTALL_PREFIX:PATH=${CMAKE_CURRENT_BINARY_DIR}/prometheus-cpp/install
 )
 
+#
+# Build crc32c
+#
+ExternalProject_Add(crc32c
+  PREFIX crc32c
+  GIT_REPOSITORY "https://github.com/google/crc32c.git"
+  GIT_TAG "1.0.6"
+  SOURCE_DIR "${CMAKE_CURRENT_BINARY_DIR}/crc32c/src/crc32c"
+  CMAKE_CACHE_ARGS
+    -DBUILD_SHARED_LIBS:STRING=yes
+    -DCRC32C_BUILD_TESTS:BOOL=OFF
+    -DCRC32C_BUILD_BENCHMARKS:BOOL=OFF
+    -DCMAKE_POSITION_INDEPENDENT_CODE:BOOL=ON
+    -DCMAKE_INSTALL_PREFIX:PATH=${CMAKE_CURRENT_BINARY_DIR}/crc32c/install
+)
+
+set(CMAKE_PREFIX_PATH ${CMAKE_PREFIX_PATH} 
+   ${CMAKE_CURRENT_BINARY_DIR}/c-ares/lib/cmake/c-ares
+   ${CMAKE_CURRENT_BINARY_DIR}/grpc/lib/cmake/grpc
+   ${CMAKE_CURRENT_BINARY_DIR}/
+   ${_FINDPACKAGE_PROTOBUF_CONFIG_DIR})
+
+#
+# Build google-cloud-cpp
+#
+ExternalProject_Add(google-cloud-cpp
+  PREFIX google-cloud-cpp
+  GIT_REPOSITORY "https://github.com/googleapis/google-cloud-cpp.git"
+  GIT_TAG "v0.10.x"
+  SOURCE_DIR "${CMAKE_CURRENT_BINARY_DIR}/google-cloud-cpp/src/google-cloud-cpp"
+  CMAKE_CACHE_ARGS
+    -DGOOGLE_CLOUD_CPP_DEPENDENCY_PROVIDER:STRING=package
+    -DGOOGLE_CLOUD_CPP_PROTOBUF_PROVIDER:STRING=package
+    -DBUILD_TESTING:BOOL=OFF
+    -DBUILD_SHARED_LIBS:BOOL=ON
```

Comment:
It would be worth trying with static libraries to see what happens... that would save us having to install the extra shared libraries.